src/main/java/de/zalando/aruha/nakadi/validation/EventBodyMustRespectSchema.java

```diff
@@ -31,8 +31,10 @@
     @Override
     public EventValidator materialize(final EventType eventType, final ValidationStrategyConfiguration vsc) {
 
-        final JSONSchemaValidator defaultSchemaValidator = new JSONSchemaValidator(
-                new JSONObject(eventType.getSchema().getSchema()));
+        JSONObject effectiveSchema = EventValidation.effectiveSchema(eventType);
```

Comment:
I agree with you that there should be a check. But I would argue that at this point, there is no useful feedback we could provide to the user.

I think the check should be done on the registry of the event type. Since we did not had this sort of checking before, this would require us to write a "clean up" routine in the database, as well. So I thought about not doing anything in this task. Wdyt @v-stepanov @CyberDem0n @jkliff ?